tests/transports/test_all_plugins.py

```diff
@@ -365,278 +301,242 @@ def test_dir_reading_permissions(custom_transport):
+        local_file_abs_path = str(local_dir / directory / local_file_name)
+        retrieved_file_abs_path = str(local_dir / directory / retrieved_file_name)
+        remote_file_abs_path = str(remote_dir / directory / remote_file_name)
 
         text = 'Viva Verdi\n'
-        with open(local_file_name, 'w', encoding='utf8') as fhandle:
+        with open(local_file_abs_path, 'w', encoding='utf8') as fhandle:
             fhandle.write(text)
 
         # localpath is an empty string
         # ValueError because it is not an abs path
         with pytest.raises(ValueError):
-            transport.put('', remote_file_name)
+            transport.put('', remote_file_abs_path)
         with pytest.raises(ValueError):
-            transport.putfile('', remote_file_name)
+            transport.putfile('', remote_file_abs_path)
 
         # remote path is an empty string
         with pytest.raises(OSError):
-            transport.put(local_file_name, '')
+            transport.put(local_file_abs_path, '')
         with pytest.raises(OSError):
-            transport.putfile(local_file_name, '')
+            transport.putfile(local_file_abs_path, '')
 
-        transport.put(local_file_name, remote_file_name)
+        transport.put(local_file_abs_path, remote_file_abs_path)
         # overwrite the remote_file_name
-        transport.putfile(local_file_name, remote_file_name)
+        transport.putfile(local_file_abs_path, remote_file_abs_path)
 
         # remote path is an empty string
         with pytest.raises(OSError):
-            transport.get('', retrieved_file_name)
+            transport.get('', retrieved_file_abs_path)
         with pytest.raises(OSError):
-            transport.getfile('', retrieved_file_name)
+            transport.getfile('', retrieved_file_abs_path)
 
         # local path is an empty string
         # ValueError because it is not an abs path
         with pytest.raises(ValueError):
-            transport.get(remote_file_name, '')
+            transport.get(remote_file_abs_path, '')
         with pytest.raises(ValueError):
-            transport.getfile(remote_file_name, '')
+            transport.getfile(remote_file_abs_path, '')
 
         # TODO : get doesn't retrieve empty files.
         # Is it what we want?
-        transport.get(remote_file_name, retrieved_file_name)
+        transport.get(remote_file_abs_path, retrieved_file_abs_path)
         # overwrite retrieved_file_name
-        transport.getfile(remote_file_name, retrieved_file_name)
-
-        os.remove(local_file_name)
-        transport.remove(remote_file_name)
-        # If it couldn't end the copy, it leaves what he did on
-        # local file
-        assert 'file_retrieved.txt' in transport.listdir('.')
-        os.remove(retrieved_file_name)
+        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
+        assert 'file_retrieved.txt' in [p.name for p in (local_dir / directory).iterdir()]
```

Comment:
```suggestion
```

Reply:
I change it to:

```python
        transport.get(remote_file_abs_path, retrieved_file_abs_path)
        assert Path(retrieved_file_abs_path).exists()
        t1 = Path(retrieved_file_abs_path).stat().st_mtime_ns


        # overwrite retrieved_file_name in 0.01 s
        time.sleep(0.01)
        transport.getfile(remote_file_abs_path, retrieved_file_abs_path)
        assert Path(retrieved_file_abs_path).exists()
        t2 = Path(retrieved_file_abs_path).stat().st_mtime_ns

        # Check st_mtime_ns to sure it is override
        assert t2 > t1
```

To make sure it is override by the second operation of `getfile`.